python/ray/train/backends/backend.py

```diff
@@ -390,7 +350,7 @@ def train_async():
 
         self.worker_group.execute_async(train_async)
 
-    def _get_next_results(self) -> Optional[List[TrainingResult]]:
+    def get_next_results(self) -> Optional[List[TrainingResult]]:
         """Fetches the next ``TrainingResult`` from each worker.
 
         Each ``TrainingResult`` is expected to correspond to the same step from
```

Comment:
Yeah I think we probably need to think more about these error messages since they're at `BackendExecutor` and not directly tied to user-facing `Trainer` APIs.